datafusion/optimizer/src/simplify_expressions/expr_simplifier.rs

```diff
@@ -365,6 +367,32 @@ impl<'a, S: SimplifyInfo> ExprRewriter for Simplifier<'a, S> {
                     None => lit_bool_null(),
                 }
             }
+            Expr::InList {
+                expr: _expr,
+                list,
+                negated,
+            } if list.is_empty() => lit(negated),
```

Comment:
I don't know how important this is, but this only holds if `expr` is non null 🤔 

```
postgres=# select null in (1);
 ?column? 
----------
 
(1 row)
```

However, note that you can't even make an empty inlist in postgres so I don't know how much this edge case matters

```sql
postgres=# select null in ();
ERROR:  syntax error at or near ")"
LINE 1: select null in ();
```